File: go-client-sdk/pkg/models/operations/getcompaniescompanyiddatahistorydatasetid.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetCompaniesCompanyIDDataHistoryDatasetIDPathParams struct {
	CompanyID string `pathParam:"style=simple,explode=false,name=companyId"`
	DatasetID string `pathParam:"style=simple,explode=false,name=datasetId"`
}

type GetCompaniesCompanyIDDataHistoryDatasetIDSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetCompaniesCompanyIDDataHistoryDatasetIDRequest struct {
	PathParams GetCompaniesCompanyIDDataHistoryDatasetIDPathParams
	Security   GetCompaniesCompanyIDDataHistoryDatasetIDSecurity
}

type GetCompaniesCompanyIDDataHistoryDatasetIDResponse struct {
	CodatPublicAPIModelsDataDataSet *shared.CodatPublicAPIModelsDataDataSet
	ContentType                     string
	StatusCode                      int
}

File: typescript-client-sdk/src/sdk/models/operations/getcompaniescompanyiddatahistorydatasetid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetCompaniesCompanyIdDataHistoryDatasetIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=datasetId" })
  datasetId: string;
}


export class GetCompaniesCompanyIdDataHistoryDatasetIdSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeApiKeyAuth;
}


export class GetCompaniesCompanyIdDataHistoryDatasetIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCompaniesCompanyIdDataHistoryDatasetIdPathParams;

  @SpeakeasyMetadata()
  security: GetCompaniesCompanyIdDataHistoryDatasetIdSecurity;
}


export class GetCompaniesCompanyIdDataHistoryDatasetIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatPublicApiModelsDataDataSet?: shared.CodatPublicApiModelsDataDataSet;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
